modules/reindex/src/main/java/org/opensearch/index/reindex/Reindexer.java

```diff
@@ -240,6 +240,9 @@ static RestClient buildRestClient(RemoteInfo remoteInfo, ReindexSslConfig sslCon
                 request.getDestination().versionType() != VersionType.INTERNAL,
                 false, logger, client, threadPool, request, listener, scriptService, sslConfig);
             this.interceptor = interceptor;
+            if (mainRequest.getRemoteInfo() != null) {
```

Comment:
I am wondering, you can initialize at one place only instead of having an if check and reload

Reply:
I can pass the `interceptor` argument to the super constructor `AsyncIndexBySearchAction`. Then I do not need to invoke `buildScrollableResultSource` again. But then would have to change reference of `AsyncIndexBySearchAction` to pass empty `interceptor` from other sub classes or create a wrapper constructor which passes empty `interceptor` to the constructor with the extra argument. Does that sound fine ?

Reply:
passing it in constructor seems like the right option. yeah, you can create the wrapper constructor to pass if other sub classes don't need the interceptor.